File: javascript/bob/bob.js
/* eslint-disable no-unused-vars */
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const hey = (message) => {
  const msg = message.trim();
  const isShouting = msg === msg.toUpperCase();
  const isQuestion = msg.charAt(msg.length - 1) === "?";
  const isNumber = !msg.match(/[a-zA-Z]+/g);

  if(msg.length === 0) return "Fine. Be that way!";
  if (!isQuestion && isShouting && !isNumber) return "Whoa, chill out!";
  if (isQuestion && !isShouting || isNumber && isQuestion) return "Sure.";
  if (isQuestion && isShouting && !isNumber) return "Calm down, I know what I'm doing!";

  return "Whatever.";
};

File: ruby/bob/bob.rb
class Bob
  class << self
    def hey(r)
      remark = r.strip
      is_shouting = remark == remark.upcase && remark.count("a-zA-Z") > 0
      is_a_question = remark.end_with?('?')
      is_empty = remark.length.zero?

      return 'Fine. Be that way!' if is_empty
      return 'Calm down, I know what I\'m doing!' if is_a_question && is_shouting
      return 'Whoa, chill out!' if is_shouting
      return 'Sure.' if is_a_question

      'Whatever.'
    end
  end
end